doc/dev/sharp_bits.rst

```diff
@@ -0,0 +1,1015 @@
+-----------------------------
+
+Inside of a qjit-compiled function, JAX transformations
+(``jax.grad``, ``jax.jacobian``, ``jax.vmap``, etc.)
+can be used **as long as they are not applied to quantum processing**.
+
+>>> @qjit
+... def f(x):
+...     def g(y):
+...         return -jnp.sin(y) ** 2
+...     return jax.grad(g)(x)
+>>> f(0.4)
+array(-0.71735609)
+
+If they are applied to quantum processing, an error will occur:
+
+>>> @qjit
+... def f(x):
+...     @qml.qnode(dev)
+...     def g(y):
+...         qml.RX(y, wires=0)
+...         return qml.expval(qml.PauliX(0))
+...     return jax.grad(lambda y: g(y) ** 2)(x)
+>>> f(0.4)
+NotImplementedError: must override
+
+Instead, only Catalyst transformations will work when applied to hybrid
+quantum-classical processing:
+
+>>> @qjit
+... def f(x):
+...     @qml.qnode(dev)
+...     def g(y):
+...         qml.RX(y, wires=0)
+...         return qml.expval(qml.PauliZ(0))
+...     return grad(lambda y: g(y) ** 2)(x)
+>>> f(0.4)
+array(-0.71735609)
+
+Always use the equivalent Catalyst transformation
+(:func:`catalyst.grad`, :func:`catalyst.jacobian`, :func:`catalyst.vjp`, :func:`catalyst.jvp`)
+inside of a qjit-compiled function.
+
+Inspecting and drawing circuits
+-------------------------------
+
+A useful tool for debugging quantum algorithms is the ability to draw them. Currently,
+:func:`@qjit <~.qjit>` compiled QNodes used as input to
+:func:`qml.draw <pennylane.draw>`, with the following caveats:
+
+- :func:`qml.draw <pennylane.draw>` call must occur outside the :func:`@qjit <.qjit>`
+
+- The :func:`@qjit <.qjit>` decorator must be placed directly on top of the QNode
+
+- The :func:`catalyst.measure` function is not supported in drawn QNodes
+
+- Catalyst conditional functions, such as :func:`~.cond` and
+  :func:`~.for_loop`, will be 'unrolled'. That is, the drawn circuit will
+  be a straight-line circuit, without any of the control flow represented
+  explicitly.
+
+For example,
+
+.. code-block:: python
+
+    @qjit
+    @qml.qnode(dev)
+    def circuit(x):
+        def measurement_loop(i, y):
+            qml.RX(y, wires=0)
+            qml.RY(y ** 2, wires=1)
+            qml.CNOT(wires=[0, 1])
+
+            @cond(y < 0.5)
+            def cond_gate():
+                qml.CRX(y * jnp.exp(- y ** 2), wires=[0, 1])
+
+            cond_gate()
+
+            return y * 2
+
+        for_loop(0, 3, step=1)(measurement_loop)(x)
+        return qml.expval(qml.PauliZ(0))
+
+>>> print(qml.draw(circuit)(0.3))
+0: ──RX(0.30)─╭●─╭●─────────RX(0.60)─╭●──RX(1.20)─╭●─┤  <Z>
+1: ──RY(0.09)─╰X─╰RX(0.27)──RY(0.36)─╰X──RY(1.44)─╰X─┤
+
+At the moment, additional PennyLane `circuit inspection functions
+<https://docs.pennylane.ai/en/stable/introduction/inspecting_circuits.html>`__
+are not supported with Catalyst.
+
+Classical control debugging
+---------------------------
+
+.. note::
+
+    See our AutoGraph guide for converting native Python control flow
+    to QJIT compatible control.
```

Comment:
I want to avoid the notion that this might be manual process 😅 
```suggestion
    See our AutoGraph guide for seamless conversion of native Python control flow
    to QJIT compatible control flow.
```